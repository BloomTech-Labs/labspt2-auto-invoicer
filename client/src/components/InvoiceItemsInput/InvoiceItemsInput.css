.invoice-item-container {
  /* border: 1px solid peru; */
  margin-top: 10px;
  margin-bottom: 10px;
  display: flex;
  justify-content: space-around;
}
