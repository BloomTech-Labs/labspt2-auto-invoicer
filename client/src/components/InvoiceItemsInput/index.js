import React from "react";

import "./InvoiceItemsInput.css";

const InvoiceItemsInput = props => {
  return props.invoiceItems.map((val, idx) => {
    let itemId = `item-${idx}`,
      quantityId = `quantity-${idx}`,
      rateId = `rate-${idx}`,
      amountId = `amount-${idx}`;
    return (
      <div key={idx} className="invoice-item-container">
        <div>
          <label htmlFor={itemId}>{`Item #${idx + 1}`}</label>
          <input
            type="text"
            name={itemId}
            data-id={idx}
            id={itemId}
            value={props.invoiceItems[idx].name}
            className="item"
            placeholder={"Item"}
          />
        </div>
        <div>
          <label htmlFor={quantityId}>{`Quantity #${idx + 1}`}</label>
          <input
            type="text"
            name={itemId}
            data-id={idx}
            id={itemId}
            value={props.invoiceItems[idx].name}
            className="quantity"
          />
        </div>
        <div>
          <label htmlFor={rateId}>{`Rate #${idx + 1}`}</label>
          <input
            type="text"
            name={itemId}
            data-id={idx}
            id={itemId}
            value={props.invoiceItems[idx].name}
            className="rate"
          />
        </div>
        <div>
          <label htmlFor={amountId}>{`Amount #${idx + 1}`}</label>
          <input
            type="text"
            name={itemId}
            data-id={idx}
            id={itemId}
            value={props.invoiceItems[idx].name}
            className="amount"
          />
        </div>
      </div>
    );
  });
};

export default InvoiceItemsInput;
