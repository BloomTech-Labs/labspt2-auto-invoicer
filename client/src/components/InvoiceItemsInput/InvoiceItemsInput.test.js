import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import InvoiceItemsInput from "../InvoiceItemsInput";

describe("<InvoiceItemsInput />", () => {
  it("renders Invoice Items Input", () => {
    const { asFragment } = render(
      <Router>
        <InvoiceItemsInput
          invoiceItems={[{ item: "test", quantity: 2, rate: "1.00" }]}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
