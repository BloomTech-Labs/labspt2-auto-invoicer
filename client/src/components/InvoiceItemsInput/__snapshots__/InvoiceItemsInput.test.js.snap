// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvoiceItemsInput /> renders Invoice Items Input 1`] = `
<DocumentFragment>
  <div
    class="invoice-item-container"
  >
    <div>
      <label
        for="item-0"
      >
        Item #1
      </label>
      <input
        class="item"
        data-id="0"
        id="item-0"
        name="item-0"
        placeholder="Item"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="quantity-0"
      >
        Quantity #1
      </label>
      <input
        class="quantity"
        data-id="0"
        id="item-0"
        name="item-0"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="rate-0"
      >
        Rate #1
      </label>
      <input
        class="rate"
        data-id="0"
        id="item-0"
        name="item-0"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="amount-0"
      >
        Amount #1
      </label>
      <input
        class="amount"
        data-id="0"
        id="item-0"
        name="item-0"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
